lib/dockerregistry/inventory.go

```diff
@@ -232,7 +241,22 @@ func ParseManifestsFromDir(
 			return nil
 		}
 
-		mfest, errParse := parseManifestFunc(path)
+		// If there are any files named "promoter-manifest.yaml", they must be
+		// inside a subfolder within "manifests/<dir>" --- any other paths are
+		// forbidden.
+		shortened := strings.TrimPrefix(path, dir)
+		shortenedList := strings.Split(shortened, "/")
+		if len(shortenedList) != ThinManifestDepth {
+			return fmt.Errorf("unexpected manifest path %q",
+				path)
+		}
+
+		if !reflect.DeepEqual(shortenedList[0:2], []string{"", "manifests"}) {
```

Comment:
Nit: This might be overkill for reflect!  Also, if we made sure that `path` always had a `/` at the end would we avoid the first "" ?